core/src/main/scala/org/dbpedia/extraction/mappings/DraftMappingExtractor.scala

```diff
@@ -0,0 +1,97 @@
+
+    val propertyParserFuncions = parserFunctions.filter(p => (p.title.equalsIgnoreCase("#property") && p.children.nonEmpty && !p.children.head.toString.contains("from")))
+    val propertyParserFuncionsHints = propertyParserFuncions.map(_.children.head.toString)
+    val propertyParserFuncionsMappings = getTemplateMappingsFromPropertyParserFunc(propertyParserFuncions)
+
+
+    val invokeFunc = parserFunctions.filter(p => p.title.equalsIgnoreCase("#invoke"))
+    val wikidataParserFunc = invokeFunc.filter(p => p.children.headOption.get.toPlainText.toLowerCase.startsWith("wikidata"))
+    val propertyLinkParserFunc = invokeFunc.filter(p => p.children.headOption.get.toPlainText.toLowerCase.startsWith("propertyLink"))
+
+    val mappingQuads = propertyParserFuncionsMappings.map( p => {
+      val value = p._1.toString + "=>" + p._2.toString
+      new Quad(context.language, mapDataset, subjectUri, templateParameterProperty,
+        value, page.sourceUri, context.ontology.datatypes("xsd:string")) })
+
+    val parserFuncQuads = (propertyParserFuncions ++ wikidataParserFunc ++ propertyLinkParserFunc).map( p =>
+      new Quad(context.language, hintDataset, subjectUri, templateParameterProperty,
+        p.toWikiText, page.sourceUri, context.ontology.datatypes("xsd:string"))
+    )
+
+    val templateQuads = ExtractorUtils.collectTemplatesFromNodeTransitive(page)
+      .filter(t => List("conditionalurl",/* "official_website",*/ "wikidatacheck").contains(t.title.encoded.toString.toLowerCase))
+      .map(t => new Quad(context.language, hintDataset, subjectUri, templateParameterProperty,
+        t.toWikiText, page.sourceUri, context.ontology.datatypes("xsd:string")))
+
+
+
+    parserFuncQuads ++ templateQuads ++ mappingQuads
+
+  }
+
+  def getPropertyParserFunctions(page : PageNode, subjectUri : String) : List[Quad] = {
+    val parserFunctions = ExtractorUtils.collectParserFunctionsFromNode(page)
+
+    val propertyParserFuncions = parserFunctions.filter(p => (p.title.equalsIgnoreCase("#property") && p.children.nonEmpty && !p.children.head.toString.contains("from")))
+    val propertyParserFuncionsHints = propertyParserFuncions.map(_.children.head.toString)
+
+    val parserFuncQuads = (propertyParserFuncions ).map( p =>
+      new Quad(context.language, hintDataset, subjectUri, templateParameterProperty,
+        p.toWikiText, page.sourceUri, context.ontology.datatypes("xsd:string"))
+    )
+
+    parserFuncQuads
+  }
+
+  private def getTemplateMappingsFromPropertyParserFunc(propertyFunctions: Seq[ParserFunctionNode]) : Seq[(String, String)] = {
+
+    for { p <- propertyFunctions;
+          if (p.parent != null && p.parent.children.size >= 2);
+          parameterSiblings = ExtractorUtils.collectTemplateParametersFromNode(p.parent);
+          if (parameterSiblings.size == 1)
+
+
+    } yield (parameterSiblings.head.parameter -> p.children.head.toPlainText)
+
+  }
+
+  def getPropertyTuples(node: Node) : String= {
+    ""
+  }
+
```

Comment:
as noted below (https://github.com/dbpedia/extraction-framework/pull/459/files#r64334141)

it is better if you create a static function here that takes a `page : PageNode` as input and returns a Tuple3,
Then, in these unit tests focus on this function only